### DISCUSSION TITLE
Themes not applying

### DISCUSSION THREAD

**Discussion by Boh132Boh** [ErikReider/SwayNotificationCenter#294] [Q&A] [answered]
I copied [these dotfiles](https://github.com/lvntcnylmz/dotfiles/tree/main/.config/swaync)  in /home/user/.config/swaync/... and used swaync-client --reload-config, but the notification center didn't change. So I tried to stop and restart swaync, but nothing changed. Then I tried restarting the computer, recompiling and reinstalling following [Debian's guide for backporting](https://wiki.debian.org/SimpleBackportCreation) and tried reloading the css and cofig files in any way I knew and was suggested, but I couldn't manage to get the theme applied.

For context I'm using Debian 12 "Bookworm" and compiled the latest version on my own since I wish to use features like hte volume and brightness control and the version in the repos doesn't support those. 

I already got some [help on reddit](https://www.reddit.com/r/swaywm/comments/14qsn8s/themes_not_applying_in_sway_notification_center/), but we couldn't find out how to solve. 

Attached a screen recording:
https://github.com/ErikReider/SwayNotificationCenter/assets/82704295/5019352d-9785-4068-84e0-eab3f856ed61

Any idea on how to solve? Thank you in advance!

**Comment by ErikReider** [comment]
Very strangle... Have you installed the swaync package? Use `dpkg -L sway-notification-center | grep ".css"`

**Reply by Boh132Boh** [reply]
Yes, I've installed it.

Here is the output of the command:
```
user@Debian:~$ dpkg -L sway-notification-center | grep ".css"
/etc/xdg/swaync/style.css
user@Debian:~$ swaync-client --reload-config
user@Debian:~$ dpkg -L sway-notification-center | grep ".css"
/etc/xdg/swaync/style.css
user@Debian:~$ swaync-client -R -rs
user@Debian:~$ dpkg -L sway-notification-center | grep ".css"
/etc/xdg/swaync/style.css
```

**Comment by ErikReider** [comment]
The latest commit [e400600](https://github.com/ErikReider/SwayNotificationCenter/commit/e400600d9c9a84fba0345192da89c278fbba46d2) will tell you which config/css files are being loaded

**Reply by Boh132Boh** [reply]
Downloaded, compiled and installed. Here is what I get:
```
$ swaync -c /home/user/.config/swaync/config.json -s /home/user/.config/swaync/style.css
** Message: 14:38:16.489: factory.vala:42: Loading widget: inhibitors
** Message: 14:38:16.489: factory.vala:42: Loading widget: title
** Message: 14:38:16.489: factory.vala:42: Loading widget: dnd
```

And in an other terminal window:
```
$ dpkg -L sway-notification-center | grep ".css"
/etc/xdg/swaync/style.css
```



I'm beginning to think I do something wrong when I compile...
My steps are:
`sudo apt autoremove --purge sway-notification-center` To remove the old installation
`git clone https://github.com/ErikReider/SwayNotificationCenter.git`
_rename the folder_
`cd /home/user/sources/sway-notification-center-0.9.0/`
`meson build`
`ninja -C build`
`dh_make --createorig`
`dh_auto_configure --buildsystem=meson`
`dpkg-buildpackage -rfakeroot -us -uc -b`
`cd /home/user/sources/`
`sudo dpkg -i sway-notification-center_0.9.0-1_amd64.deb`

Note: I'm not a developer and I don't know my way around compiling, I just followed the instructions in the readme and a guide I found online.

**Reply by ErikReider** [reply]
You can try to run the compiled executable without installing the package. Inside of the repo directory, run `ninja -C build && ./build/src/swaync`. This is how I run it when developing :)

**Reply by Boh132Boh** [reply]
It didn't work, seems to load only the wrong config...

Terminal 1:
```
user@Debian:~/.config/swaync$ ls
a-config.json  a-style.css  b-config.json  b-style.css  config.json  style.css
user@Debian:~/.config/swaync$ cd ~/sources/sway-notification-center-0.9.0/
user@Debian:~/sources/sway-notification-center-0.9.0$ ninja -C build && ./build/src/swaync
ninja: Entering directory `build'
ninja: no work to do.
** Message: 12:29:10.231: configModel.vala:315: Loading Config: "/usr/local/etc/xdg/swaync/config.json"
** Message: 12:29:10.232: functions.vala:78: Loading CSS: "/usr/local/etc/xdg/swaync/style.css"
** Message: 12:29:10.233: functions.vala:92: Loading CSS: "/usr/local/etc/xdg/swaync/style.css"
** Message: 12:29:10.286: factory.vala:42: Loading widget: inhibitors
** Message: 12:29:10.290: factory.vala:42: Loading widget: title
** Message: 12:29:10.291: factory.vala:42: Loading widget: dnd
```
Terminal 2:
```
user@Debian:~$ swaync-client -R -rs
```
Terminal 1:
```
** Message: 12:29:32.259: swayncDaemon.vala:180: Reloading config

** Message: 12:29:32.259: configModel.vala:315: Loading Config: "/usr/local/etc/xdg/swaync/config.json"
** Message: 12:29:32.263: factory.vala:42: Loading widget: inhibitors
** Message: 12:29:32.263: factory.vala:42: Loading widget: title
** Message: 12:29:32.264: factory.vala:42: Loading widget: dnd
```

_**Stops swaync**_
Terminal 1:
```
user@Debian:~/sources/sway-notification-center-0.9.0$ ninja -C build && ./build/src/swaync -c ~/.config/swaync/config.json -s ~/.config/swaync/style.css 
ninja: Entering directory `build'
ninja: no work to do.
** Message: 12:34:59.465: configModel.vala:315: Loading Config: "/usr/local/etc/xdg/swaync/config.json"
** Message: 12:34:59.466: functions.vala:78: Loading CSS: "/usr/local/etc/xdg/swaync/style.css"
** Message: 12:34:59.468: functions.vala:92: Loading CSS: "/usr/local/etc/xdg/swaync/style.css"
** Message: 12:34:59.519: factory.vala:42: Loading widget: inhibitors
** Message: 12:34:59.520: factory.vala:42: Loading widget: title
** Message: 12:34:59.520: factory.vala:42: Loading widget: dnd
```

Terminal 2:
```
user@Debian:~$ swaync-client -R -rs
```

Terminal 1:
```
** Message: 12:35:23.135: swayncDaemon.vala:180: Reloading config

** Message: 12:35:23.136: configModel.vala:315: Loading Config: "/usr/local/etc/xdg/swaync/config.json"
** Message: 12:35:23.139: factory.vala:42: Loading widget: inhibitors
** Message: 12:35:23.140: factory.vala:42: Loading widget: title
** Message: 12:35:23.140: factory.vala:42: Loading widget: dnd
```

**Reply by ErikReider** [reply, marked as answer]
Oh I know what's wrong... Moving this into issue #295 :)
